services/galley/src/Galley/API/Teams.hs

```diff
@@ -475,8 +484,16 @@ deleteTeamMember zusr zcon tid remove mBody = do
     then do
       body <- mBody & ifNothing (invalidPayload "missing request body")
       ensureReAuthorised zusr (body ^. tmdAuthPassword)
+      (TeamSize sizeBeforeDelete) <- BrigTeam.getSize tid
+      -- TeamSize is 'Natural' and subtracting from  0 is an error
+      -- TeamSize could be reported as 0 if team members are added and removed very quickly,
+      -- which happens in tests
+      let sizeAfterDelete =
+            if sizeBeforeDelete == 0
+              then 0
+              else sizeBeforeDelete - 1
```

Comment:
hum, this does not inspire a lot of trust...  but i have no idea how to make this better.  :(

so if the same thing happens between 1809 and 1810 instead of 0 and 1, the result will be off by one?

Reply:
I don't understand what you mean by 1809 and 1810. Can you please explain?

Reply:
i mean that if you run into this race condition with number higher than 1 and 0, this will just silently get the count wrong by 1.

Does this accumulate?  So every time we enter the count becomes less accurate?

Reply:
Yes, this can be off by more than 1, if a lot of people join/leave the team within a very short time (seconds), we are bound to get this number wrong.